### ISSUE TITLE
关于binary reward，grad_accumulate, cfg的问题。

### ISSUE THREAD

**Issue by colorful-liyu** [XueZeyue/DanceGRPO#35]
作者您好！非常感谢开源这么棒的工作！对我们的项目帮助很大，我们正在following您。
但在实现上我们有几个疑问。

1. 论文在Sec.3.2提到使用了a binary reward formulation，但是在开源代码中并没有找到这部分实现。想请教一下这部分的作用如何？具体是怎么样实现的呢？
2. 我看到本文的实现并没有使用cfg，但是flow-grpo中使用了neg_prompt主导的cfg，想请教您，使用cfg与不使用cfg在效果上有多大差异呢？当时您是如何考量取舍的。
3. 在给出的finetune_flux_grpo.sh脚本中，使用了gradient_accumulation_steps=4，不能支持对同一批采样的图片进行完整训练后再更新模型。但是在flow-grpo的代码中，使用了较大的gradient_accumulation_steps。请问这方面的选择有什么说法吗？

再次感谢您的项目！

**Comment by XueZeyue** [comment]
1. binary reward:
加一行代码即可，在那个计算hps_score的后面:

hps_score = torch.tensor([1.0]).to(device) if hps_score > 0.28 else torch.tensor([0.0]).to(device)

hps训的很快，clip会慢一点，然后阈值要重新设置，可能0.39，当然这个是验证型实验，就想看看grpo能不能学sparse的分布，所以论文没有over claim，只说了initial attempts

2. cfg:
能不带cfg就别带，我所有带cfg的setting都更不稳定，所以我flux和hunyuanvideo都没带，sd和skyreels-i2v是没办法，必须开，我甚至遇到过带cfg训不出来的情况，关掉就好了

3. 梯度累积
这个实话实说，我觉得不开cfg，调多大都没区别，开cfg要调大一点，我i2v，一个batch采样prompt只更新一次梯度，因为更新多了会发散

**Comment by colorful-liyu** [comment]
感谢您的回复，非常有帮助也非常及时！
 这边还有几个细节想进一步确认一下：

1. binary reward与普通的连续reward在效果上有明显差异吗？另外就是，我在实验中发现如果使用binary reward，可能会让某个组内部的reward全为正或者全为负，这种情况您有观察到吗？
2. 为什么sd和skyreels-i2v必须使用cfg呢？我注意到实验中使用了guidance=3.5，而cfg却需要关掉，这个现象有点让人困惑。

**Comment by XueZeyue** [comment]
1. 我没感觉有啥差异，可能我这个实验本身就是论文里所说的initial attempts，当时做这个实验也是想启发社区能不能真正构造出视觉的rule-based reward；对，这种情况会有，所以要调参

2. 因为这俩模型不开cfg出不了正常的图/视频，reward model无法感知，你说的这个3.5应该是cfg-distilled的超参，就是模拟cfg的结果，不是真正的cfg

**Comment by colorful-liyu** [comment]
感谢您的答疑解惑！我按您的指导进行实验，后面有新结果再来分享！

**Comment by ICE-Bruce** [comment]
请问开CFG后，概率的计算需要修改吗？
